### DISCUSSION TITLE
The Hive: Created by / Assignee name Issue

### DISCUSSION THREAD

**Discussion by northernrevival** [Security-Onion-Solutions/securityonion#4153] [Unsupported Versions]
We have come across an in issue in The Hive that was previously mentioned by others but not solved. We have many accounts for users to log into the Security Onion console to triage as they need. The problem arises when escalating alerts to The Hive, regardless of the specific user who actually escalated the ticket:

- The 'Created by' for the case defaults to a non-personal sort of maintenance account we have set up
- The 'Assignee' also does the same, defaulting to the same maintenance account
- We know for sure that before we experienced this problem a group of cases were definitely both assigned to specific individuals when created, and also created by specific individuals, but looking back these have also defaulted to the same maintenance account

This could be a misconfiguration on our part, and knowing that at least one other user has posted on this space I'm guessing there's a few more of us out there that are dealing/have dealt with it.

Any help or information on this subject would be greatly appreciated.

**Comment by cyberkimh** [comment]
Yes we are having this issue also. The assignee is the same name no matter who escalates the alert and creates the case.

**Reply by Stephen-Ridgway** [reply]
I've spent some time looking at this, but I can't work out how TheHive knows which user is logged in to SOC.

In global.sls there are entries for 'hivekey:' and 'hiveplaysecret:'. I'm guessing that hiveplaysecret is the glue between playbook and hive, and hivekey is the API key for the hive.

As the hivekey value is set in global.sls to be the API key for a service account (in our configuration), it looks like all alerts escalated will be created with that service account's credentials. This is what happens in practice.

Is it possible to change this behaviour? If so, how?

**Reply by Stephen-Ridgway** [reply]
Further updates:

If I remove the API key value from global.sls, SOC no longer offers me the option to escalate alerts. "escalateEnabled:" becomes 'false'

If I put back the API key of the service account, I get:


                  @@ -38,6 +38,12 @@
                           "offlineThresholdMs": 900000
                         },
                   
                  +      "thehive": {
                  +        "hostUrl": "http://a.b.c.d:9000/thehive",
                  +        "key": “xxxxxxxx”,
                  +        "verifyCert": false
                  +      },
                  +
                         "statickeyauth": {
                           "anonymousCidr": “a.b.0.0/24",
                           "apiKey": “yyyyyyy”
                  @@ -64,7 +70,7 @@
                           "relativeTimeUnit": 30,
                           "mostRecentlyUsedLimit": 5,
                           "ackEnabled": false,
                  -        "escalateEnabled": false,
                  +        "escalateEnabled": true,

when I restart TheHive, which reverts back to creating all cases under the service account.

As has been mentioned before, this is not the behaviour we've seen before, but I can't be certain when the behaviour changed.

Any help appreciated!

**Reply by B3DTech** [reply]
This is normal from what I see - and the way the Hive has always worked that I'm aware of.  The original account you created for TheHive is the key and therefore the "user" that escalations get added as.

**Reply by northernrevival** [reply]
Hi everyone it's nice to see people are picking this one up and looking into it - I thank you all.

If this is normal behaviour for the Hive @B3DTech then it causes problems for our deployment as we have internal analysts at the SOC and also customer analysts at the SOC, all logged in as themselves, but when cases are escalated it is impossible to know who escalated the case - and this seems to us very important to know. 

@Stephen-Ridgway I don't recall it always being this way, I'm sure it followed an update or maybe the migration from 16.04 to 2.3 but I cannot be sure. By the time we noticed what was happening - which I believe followed a few 2.3 tutorial sessions we delivered to customers - it was too late. However this is why I have the feeling that it might have followed the migration. Thank you for looking into this for us on the backend and please keep us updated on anything you find.

If the user that the Hive account was created with is the account the key is generated for, do we know if it is possible to generate further keys for other users of the SOC and therefore the Hive?

Again thank you all for your efforts, I had hit a wall when trying to figure out a) what was wrong and b) how to fix it.

**Comment by vijaytejak** [comment]
hey all, did you find any workaround for this?

**Comment by vijaytejak** [comment]
The only temporary solution i see is to manually change the assignee after the alert is escalated to Hive.

**Reply by northernrevival** [reply]
Hi @vijaytejak thank you for your interest, we didn't find a solution, as mentioned above by others The Hive is authenticated using an umbrella or admin account, and seems to have no way of knowing which user escalates from the SOC.

This being said, we have worked around it like yourself. Manually assigning tasks to individual users, good communication with those you're assigning to and assigning to yourself if you're carrying out the investigation, and proper case management providing clear descriptions of the case and what is expected of others.

It's all going smoothly so far! Would be great if there was a solution for this though, as looking back over old cases where your memory of them is patchy at best, can be difficult.

**Comment by vijaytejak** [comment]
@northernrevival Can you share the best practices you are following with Hive, i know you will be busy, pls find some time and share the best methods you follow.

Thanks in advance
